src/views/AdventureMapView.js

```diff
@@ -392,8 +393,36 @@ exports = Class(ScrollView, function (supr) {
 		if (y < 0) {
 			this.populateBottom(posY);
 		} else if (y > 0) {
-			this.populateTop(posY);
+			if(posY > toScrollMin) {
+				this.populateLargeTop(posY);
+			} else {
+				this.populateTop(posY);
+			}
+		}
+	};
+
+	this.populateLargeTop = function (count) {
+		var num = Math.floor(count / this._tileHeight),
+			end;
+
+		// Handle reaching end of map
+		if (v_tail -num < 0) {
+			num = v_tail;
 		}
+		end = v_tail - num;
+		v_slider_tail -= num;
+		v_slider_head -= num;
+		v_tail = v_slider_head + v_padding;
+
+		for (var y = v_tail - 1, i = 0; y >= end; y--, i++) {
+			var rel = v_head - i;
+			for (var x = h_tail; x < h_head; x++) {
+				this.create(x, y, x, rel);
```

Comment:
1st and 2nd param are x&y of what has to be created
3rd and 4th are x&y of what has to be released